src/javascript/Lock/LockInfo.badge.jsx

```diff
@@ -4,13 +4,20 @@ import {Badge} from '@jahia/design-system-kit';
 import {useTranslation} from 'react-i18next';
 import {useContentEditorContext} from '~/ContentEditor.context';
 
+const LOCK_TYPE_VALIDATION = 'validation';
+const LOCK_TYPE_DELETION = 'deletion';
+
 const getBadgeContent = (nodeData, t) => {
-    let lockInfoDetails = nodeData.lockInfo.details[0];
+    let lockInfoDetails = nodeData.lockInfo.details.find(detail => detail.type === LOCK_TYPE_VALIDATION || detail.type === LOCK_TYPE_DELETION);
+    if (!lockInfoDetails) {
+        lockInfoDetails = nodeData.lockInfo.details[0];
+    }
+
     let badgeContent;
-    if (lockInfoDetails.type === 'engine' || lockInfoDetails.type === 'user') {
-        badgeContent = t('content-editor:label.contentEditor.edit.action.lock.' + lockInfoDetails.type, {userName: lockInfoDetails.owner});
-    } else {
+    if (lockInfoDetails.type === LOCK_TYPE_VALIDATION || lockInfoDetails.type === LOCK_TYPE_DELETION) {
```

Comment:
Avoid to do the same test twice (you already know regarding the filtering result if your lock is deletion/validation, no need to test it again)